files/gen_sys_stats.rb

```diff
@@ -0,0 +1,252 @@
+  #
+  class GenerateSystemMetrics
+    #
+    # Initialize class
+    #
+    # @author Randell Pelak
+    #
+    # @param [integer] polling_interval Time in seconds between calls to poll the system for data.
+    # @param [integer] file_interval Tim in seconds between the creation of each output file.
+    # @param [string] metric_type cpu|memory
+    # @param [string] metrics_dir The puppet_metrics_collector output directory.
+    # @param [boolean] verbose Verbose output
+    #
+    # @return [void]
+    def initialize(polling_interval, file_interval, metric_type, metrics_dir, verbose = false)
+      @polling_interval = polling_interval
+      @file_interval = file_interval
+      @metric_type = metric_type
+      @metrics_dir = metrics_dir
+      @verbose = verbose
+
+      @hostname = %x[hostname].strip
+      puts "Hostname is: #{@hostname}" if @verbose
+      FileUtils.mkdir_p(metrics_dir) unless File.directory?(metrics_dir)
+    end
+
+    # Run sar to collect the raw system data
+    #
+    # @author Randell Pelak
+    #
+    # @return [string] raw output from sar
+    def run_sar
+      times_to_poll = ((@file_interval / @polling_interval) + 0.5).to_i
+      # sar inputs are polling interval and how many times to poll
+      comm_flags = " -r" if @metric_type =~ /memory/
+      comm = "/usr/bin/sar #{comm_flags} #{@polling_interval} #{times_to_poll}"
+      puts "sar command is: #{comm}" if @verbose
+      %x[#{comm}]
+    end
+
+    # Confirm the output from sar is valid
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] sar_output
+    #
+    # @return [boolean]
+    def sar_output_valid?(sar_output)
+      last_line = sar_output.split("\n")[-1]
+      return false unless last_line =~ /^Average:/
+      header = get_header_fields(sar_output)
+      averages = get_averages_fields(sar_output)
+      return false unless header.count == averages.count
+      true
+    end
+
+    # Gets the header fields from sar output
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] sar_output
+    #
+    # @return [array] header fields
+    def get_header_fields(sar_output)
+      arr = sar_output.split("\n")
+      arr[2].split(" ")[2..-1]
+    end
+
+    # Gets the averages fields from sar output
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] sar_output
+    #
+    # @return [array] average fields
+    def get_averages_fields(sar_output)
+      arr = sar_output.split("\n")
+      arr[-1].split(" ")[1..-1]
+    end
+
+    # Convert the inputted sar output into json
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] sar_output
+    # @param [obj] time_stamp_obj Time object to use for generating the filename
+    #
+    # @return [string] json of sar output
+    def convert_sar_output_to_json(sar_output, time_stamp_obj)
+      header = get_header_fields(sar_output)
+      averages = get_averages_fields(sar_output)
+      hostkey = @hostname.gsub('.', '-')
```

Comment:
Why change the hostname?  The key is a string, so the '.' characters should not cause any issue with Ruby or JSON.

Reply:
this was to be consistent with the rest of PMC.  The value is actually the default tag if the data gets loaded into influxdb... so maybe it cares... dunno for sure, but wanted it to be consistent.